dora/core/common/src/main/java/alluxio/membership/StaticMembershipManager.java

```diff
@@ -0,0 +1,183 @@
+  private final List<WorkerInfo> mMembers;
+
+  private final AlluxioConfiguration mConf;
+
+  /**
+   * @param conf
+   * @return StaticMembershipManager
+   * @throws IOException
+   */
+  public static StaticMembershipManager create(AlluxioConfiguration conf) throws IOException {
+    // user conf/workers, use default port
+    String workerListFile = conf.getString(
+        PropertyKey.WORKER_STATIC_MEMBERSHIP_MANAGER_CONFIG_FILE);
+    List<WorkerInfo> workers = parseWorkerAddresses(workerListFile, conf);
+    return new StaticMembershipManager(conf, workers);
+  }
+
+  /**
+   * CTOR for StaticMembershipManager.
+   * @param conf
+   * @throws IOException
+   */
+  @SuppressFBWarnings({"URF_UNREAD_FIELD"})
+  StaticMembershipManager(AlluxioConfiguration conf, List<WorkerInfo> members) {
+    mConf = conf;
+    mMembers = members;
+  }
+
+  /**
+   * Parse the worker addresses from given static config file.
+   * The static file only gives the hostname, the rest config params
+   * are inherited from given Configuration or default values.
+   * @param configFile
+   * @param conf
+   * @return list of parsed WorkerInfos
+   * @throws IOException
+   */
+  private static List<WorkerInfo> parseWorkerAddresses(
+      String configFile, AlluxioConfiguration conf) throws IOException {
+    List<WorkerNetAddress> workerAddrs = new ArrayList<>();
+    File file = new File(configFile);
+    if (!file.exists()) {
+      throw new FileNotFoundException("Not found for static worker config file:" + configFile);
+    }
+    Set<String> workerHostnames = CommandUtils.readNodeList("", configFile);
+    for (String workerHostname : workerHostnames) {
+      WorkerNetAddress workerNetAddress = new WorkerNetAddress()
+          .setHost(workerHostname)
+          .setContainerHost(Configuration.global()
+              .getOrDefault(PropertyKey.WORKER_CONTAINER_HOSTNAME, ""))
+          .setRpcPort(conf.getInt(PropertyKey.WORKER_RPC_PORT))
+          .setWebPort(conf.getInt(PropertyKey.WORKER_WEB_PORT));
+      //data port, these are initialized from configuration for client to deduce the
+      //workeraddr related info, on worker side, it will be corrected by join().
+      InetSocketAddress inetAddr;
+      if (Configuration.global().getBoolean(PropertyKey.USER_NETTY_DATA_TRANSMISSION_ENABLED))  {
+        inetAddr = NetworkAddressUtils.getBindAddress(
+            NetworkAddressUtils.ServiceType.WORKER_DATA,
+            Configuration.global());
+        workerNetAddress.setNettyDataPort(inetAddr.getPort());
+      } else {
+        inetAddr = NetworkAddressUtils.getConnectAddress(
+            NetworkAddressUtils.ServiceType.WORKER_RPC,
+            Configuration.global());
+      }
+      workerNetAddress.setDataPort(inetAddr.getPort());
+      workerAddrs.add(workerNetAddress);
+    }
+    return workerAddrs.stream()
+        .map(w -> new WorkerInfo().setAddress(w)).collect(Collectors.toList());
+  }
+
+  @Override
+  @VisibleForTesting
```

Comment:
Remove `@VisibleForTesting` in the following methods.
we just need it in the interface, only for these methods only exposed for testing

Reply:
done